src/stores/SessionStore.js

```diff
@@ -0,0 +1,61 @@
+import dis from '../dispatcher';
+import EventEmitter from 'events';
+
+/**
+ * A class for storing application state to do with the session. This is a simple flux
+ * store that listens for actions and updates its state accordingly, informing any
+ * listeners (views) of state changes via the 'update' event.
+ */
+class SessionStore extends EventEmitter {
+    constructor() {
+        super();
+
+        // Initialise state
+        this._state = {
+            cachedPassword: localStorage.getItem('mx_pass'),
+        };
+
+        dis.register(this._onAction.bind(this));
+    }
+
+    _update() {
+        // Persist state to localStorage
+        if (this._state.cachedPassword) {
+            localStorage.setItem('mx_pass', this._state.cachedPassword);
```

Comment:
hum. Having accesses to localStorage spread out across the app makes me sad - write access in particular. One of the goals of `Lifecycle` was to try and collect this sort of thing together.